azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeImageExtension.java

```diff
@@ -118,26 +121,47 @@ public Image call() throws Exception {
                            disks[1] = datadiskObject.get("name");
 
                            VirtualMachine vm = api.getVirtualMachineApi(group).get(id);
-                           String location = vm.location();
-                           final VMImage ref = VMImage.create(CUSTOM_IMAGE_PREFIX + group, CUSTOM_IMAGE_PREFIX + name, disks[0], disks[1], location, false);
+                           final VMImage ref = VMImage.create(group, storageAccountName, disks[0], disks[1], name, "custom", vm.location());
                            return imageReferenceToImage.apply(ref);
                         }
                      }
                   }
                }
                throw new UncheckedTimeoutException("Image was not created within the time limit: "
-                       + cloneTemplate.getName());
+                     + cloneTemplate.getName());
```

Comment:
`cloneTemplate` and `template` are both used here. Let's stick to the local one to avoid modifying references to an input variable.